### PR TITLE
Remove Scratch folder from users' config trees

### PR DESCRIPTION
**Pull request by jacobythwaites**
The Scratch folder is not needed as a default in users' trees.

Tests assume it is present. This PR removes those dependencies.

A bug was uncovered where changes to a user such as registration or password change failed to fire a configuration event and were lost upon node shutdown unless another change followed. This has been fixed.

### CODE REVIEW

**Comment by jacobythwaites** [comment]
Fixes https://github.com/opensparkl/sse_core/issues/50

File: apps/sse_cfg/src/sse_cfg.erl [hunk 1]
```diff
@@ -720,11 +725,22 @@ when
   Email :: string().
 
 delete_user(Email) ->
-  call(delete_user, [Email]).
+  try
+    ok =
+      call(delete_user, [Email]),
+    ok =
+      sse_cfg_event:log_change(
```
Comment (lucafavatellawork) [inline apps/sse_cfg/src/sse_cfg.erl]: I tested user deletion is persisted.

File: apps/sse_cfg/src/sse_cfg.erl [hunk 2]
```diff
@@ -733,12 +749,25 @@ when
   UserSpec :: list(tuple()).
 
 register_user(UserSpec) ->
-  call(register_user, [UserSpec]).
+  try
+    {ok, User} =
+      call(register_user, [UserSpec]),
+    ok =
+      sse_cfg_event:log_change(
```
Comment (lucafavatellawork) [inline apps/sse_cfg/src/sse_cfg.erl]: I tested new user registration is persisted.

File: apps/sse_svc/src/sse_svc_dispatcher_yapp.erl [hunk 3]
```diff
@@ -131,9 +131,9 @@ process_post(Arg, ?notify, #?USER{} = User, Data) ->
       subject = Subject} = NotifyEvent} =
     parse(Arg, User, Data),
 
-  {_r, _w, x} =
-    sse_cfg:permission_in(
-      User, parent_of(Subject)),
+  {ok, {{_r, _w, x}, _Rec}} =
+    sse_cfg:permission_record(
```
Comment (lucafavatellawork) [inline apps/sse_svc/src/sse_svc_dispatcher_yapp.erl]: Is this meant to be exported? If yes, is this meant to fail CI?
```
1> sse_cfg:permission_record(bar, baz).
** exception error: undefined function sse_cfg:permission_record/2
...

(sse1@C02Z30NAL413.local)6> rp(lists:sort(sse_cfg:module_info(exports))).
...
 {outer_resources,1},
 {permission_in,2},
 {priority_permission_in,2},
...
```

Comment (jacobythwaites) [reply apps/sse_svc/src/sse_svc_dispatcher_yapp.erl]: No idea what I was thinking. Corrected.

**Pull request merged**